File: day23_part2_2017.rs
use std::fs;

fn is_prime(n: i32) -> bool {
    for i in 2..=(n as f64).sqrt() as i32 {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    let input = fs::read_to_string("input.txt").expect("Unable to read file");
    let lines: Vec<&str> = input.trim().split('\n').collect();

    let b: i32 = 57 * 100 + 100000;
    let c: i32 = b + 17000;
    let mut h: i32 = 0;

    for x in (b..=c).step_by(17) {
        if !is_prime(x) {
            h += 1;
        }
    }

    println!("{}", h);
}

File: matlab/day23_part2_2017.m

function main()
    fid = fopen('input.txt', 'r');
    fclose(fid);

    b = 57 * 100 + 100000;
    c = b + 17000;
    h = 0;

    for x = b:17:c
        if ~isprime(x)
            h = h + 1;
        end
    end

    disp(h);
end

function result = isprime(n)
    if n < 2
        result = false;
        return;
    end
    for i = 2:sqrt(n)
        if mod(n, i) == 0
            result = false;
            return;
        end
    end
    result = true;
end
